pkg/apis/maistra/conversion/extension_providers_test.go

```diff
@@ -176,3 +177,38 @@ meshConfig:
 		},
 	}
 }
+
+func TestInvalidValueType(t *testing.T) {
+	helmValues := v1.HelmValues{}
+	if err := helmValues.UnmarshalYAML([]byte(`
+meshConfig:
+  extensionProviders:
+  - envoyExtAuthzHttp:
+      service: "test"
+      port: "80"
+`)); err != nil {
+		t.Fatalf("failed to parse helm values: %s", err)
+	}
+
+	specV1 := v1.ControlPlaneSpec{
+		Version: "v2.4",
+		Istio:   &helmValues,
+	}
+	var specV2 v2.ControlPlaneSpec
+	err := Convert_v1_ControlPlaneSpec_To_v2_ControlPlaneSpec(&specV1, &specV2, nil)
+	if err != nil {
+		t.Fatalf("failed to convert spec: %s", err)
+	}
+
+	errorMessage, found, err := specV2.TechPreview.GetString("errored.message")
+	if !found {
+		t.Fatalf("expected to find techPreview.errored.message field")
+	}
+	if err != nil {
```

Comment:
Shouldn't `err` be checked first? I guess this condition will never be true if executed in the current order.

Reply:
Yup, you're right.